FirebaseMessaging/Sources/FIRMessagingExtensionHelper.m

```diff
@@ -211,8 +211,25 @@ - (void)exportDeliveryMetricsToBigQueryWithMessageInfo:(NSDictionary *)info {
   fm_MessagingClientEventExtension eventExtension = fm_MessagingClientEventExtension_init_default;
 
   fm_MessagingClientEvent clientEvent = fm_MessagingClientEvent_init_default;
+  if (!info[kFIRMessagingSenderID]) {
```

Comment:
optional: I usually prefer using a local var in such cases like:
```
NSString *senderID = info[kFIRMessagingSenderID];
if(senderID.length == 0) {
...
}
clientEvent.project_number = (int64_t)[info[kFIRMessagingSenderID] longLongValue];
```

It makes relation between validation and usage more clear.

Reply:
what if it's not a string? I worry it might still pass and crash the protobuf.

Reply:
The validation is just an example, my main point here is that declaring a local variable is more readable and change proof than accessing a value from the dict by key. I started doing it after I had changed a key in one place but forgot to change at another by accident.